### PR TITLE
Remove CapsList type

### LINKED ISSUE #172: Remove CapsList
ℹ️ This issue is part of Epic #169 
## Objective

Apply the required changes for removing the **CapsList**.

### PR DESCRIPTION
**Pull request by EdwinGuayacan**
Fixes #172 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: lib/types/optional_caps_list.ex [hunk 1]
```diff
@@ -15,6 +15,6 @@ defmodule Kadena.Types.OptionalCapsList do
   @impl true
   def new(clist \\ nil)
   def new(nil), do: %__MODULE__{}
-  def new(%CapsList{} = clist), do: %__MODULE__{clist: clist}
+  def new([%Cap{} | _tail] = clist), do: %__MODULE__{clist: clist}
```
Comment (L-Zuluaga) [inline lib/types/optional_caps_list.ex:18]: is the `_tail` required? I mean, for this specific case, maybe it's better to use a guard? more descriptive?
```suggestion
  def new(clist) when is_list(clist), do: %__MODULE__{clist: clist}
```
What do you think? @miguelnietoa 🤔🤔

File: lib/types/signer.ex [hunk 2]
```diff
@@ -56,15 +56,7 @@ defmodule Kadena.Types.Signer do
 
   @spec validate_clist(clist :: cap_list()) :: validation()
   defp validate_clist(nil), do: {:ok, OptionalCapsList.new()}
-  defp validate_clist(%CapsList{} = clist), do: {:ok, OptionalCapsList.new(clist)}
+  defp validate_clist([%Cap{} | _tail] = clist), do: {:ok, OptionalCapsList.new(clist)}
```
Comment (L-Zuluaga) [inline lib/types/signer.ex:59]: Same as above 🤔

**Review by L-Zuluaga — APPROVED** [review]
LGTM!
I left a couple of comments that are more a suggestion, because I'm not sure if they are better alternatives that the ones you provided. Other than that nice work 🚀

File: lib/types/optional_caps_list.ex (hunk 1 above)
Comment (miguelnietoa) [reply lib/types/optional_caps_list.ex:18]: IMO it is better to keep `[%Cap{} | _tail]` as it validates that a list type containing Cap structs is provided to the `new/1` function

Comment (L-Zuluaga) [reply lib/types/optional_caps_list.ex:18]: yeah makes sense, my mistake 🙏🏻

**Pull request merged**